src/Components/Components/test/Routing/RouterTest.cs

```diff
@@ -42,72 +42,39 @@ public async Task CanRunOnNavigateAsync()
         {
             // Arrange
             var called = false;
-            async Task OnNavigateAsync(NavigationContext args)
+            Action<NavigationContext> OnNavigateAsync = async (NavigationContext args) =>
             {
                 await Task.CompletedTask;
                 called = true;
-            }
-            _router.OnNavigateAsync = OnNavigateAsync;
-
-            // Act
-            await _renderer.Dispatcher.InvokeAsync(() => _router.RunOnNavigateWithRefreshAsync("http://example.com/jan", false));
-
-            // Assert
-            Assert.True(called);
-        }
-
-        [Fact]
-        public async Task CanHandleSingleFailedOnNavigateAsync()
-        {
-            // Arrange
-            var called = false;
-            async Task OnNavigateAsync(NavigationContext args)
-            {
-                called = true;
-                await Task.CompletedTask;
-                throw new Exception("This is an uncaught exception.");
-            }
```

Comment:
Orthogonal to this change:  We should test what happens when this throws synchronously and asynchronously. For example:

```csharp
           Task OnNavigateAsync(NavigationContext args)
            {
                called = true;
                throw new Exception("This is an uncaught exception.");
                return Task.CompletedTask;
            }
```

Which throws synchronously VS the method above. (Which also throws synchronously, but the async machinery will transform it into a failed task).

For throwing asynchronously we actually need to go async

```csharp
           async Task OnNavigateAsync(NavigationContext args)
            {
                called = true;
                await Task.Delay(1000); // Or better than this, a task completion source
                throw new Exception("This is an uncaught exception.");
            }
```

Reply:
Added an E2E test for this. E2E tests work a little bit better here because there isn't a good way to attach the event callback the router component in the unit tests.